L1Trigger/Phase2L1ParticleFlow/interface/common/inversion.h

```diff
@@ -0,0 +1,63 @@
+#ifndef CC_INVERSION_H__
+#define CC_INVERSION_H__
+
+namespace l1ct {
+
+  constexpr int ceillog2(int x) { return (x <= 2) ? 1 : 1 + ceillog2((x + 1) / 2); }
+
+  template <class data_T, int N>
+  inline float real_val_from_idx(unsigned i) {
+    // Treat the index as the top N bits
+    static constexpr int NB = ceillog2(N);  // number of address bits for table
+    data_T x(0);
+    // The MSB of 1 is implicit in the table
+    x[x.width - 1] = 1;
+    // So we can use the next NB bits for real data
+    x(x.width - 2, x.width - NB - 1) = i;
+    return (float)x;
+  }
+
+  template <class data_T, int N>
+  inline unsigned idx_from_real_val(data_T x) {
+    // Slice the top N bits to get an index into the table
+    static constexpr int NB = ceillog2(N);  // number of address bits for table
+    // Slice the top-1 NB bits of the value
+    // the MSB of '1' is implicit, so only slice below that
+    ap_uint<NB> y = x(x.width - 2, x.width - NB - 1);
+    return (unsigned)y(NB - 1, 0);
+  }
+
+  template <class data_T, class table_T, int N>
+  void init_invert_table(table_T table_out[N]) {
+    // The template data_T is the data type used to address the table
+    for (unsigned i = 0; i < N; i++) {
+      float x = real_val_from_idx<data_T, N>(i);
+      table_T inv_x = 1 / x;
+      table_out[i] = inv_x;
+    }
+  }
+
+  template <class in_t, class table_t, int N>
+  table_t invert_with_shift(in_t in) {
+    table_t inv_table[N];
+    init_invert_table<in_t, table_t, N>(inv_table);
+
+    // find the first '1' in the denominator
+    int msb = 0;
+    for (int b = 0; b < in.width; b++) {
+      // #pragma HLS unroll
+      if (in[b])
+        msb = b;
```

Comment:
If msb must point to the first one, as written in the comment above, you should probably break here

Reply:
This looks correct to me (msb= Most Significant Bit). I let the original author @thesps comment further.

Reply:
Ok. Then probably "first" should be better specified in the comment line ("most significant"?).
As a possible further tiny improvement (to be taken into account in possible further update PRs), if you do the loop in the opposite direction and `break` at the first '1' you can probably save some cycle.